File: UI/Web/src/app/blocks/models/template.ts
import { ISnippet } from './snippet';


export interface ITemplate {
    id?: string | null
    name?: string | null
    snippets?: ISnippet[] | null
}

export class Template implements ITemplate {
    constructor(
        public id: string | null = null,
        public name: string | null = null,
        public width: number | null = null,
        public height: number | null = null,
        public snippets: (ISnippet | null)[]  | null = null,
    ) { }
}

File: system/Infrastructure/Infrastructure/Domain/Template.cs
using Infrastructure.Helper;
using System;
using System.Collections.Generic;

namespace Infrastructure.Domain
{
    [CollectionName("Templates")]
    public class Template : Entity
    {
        public Template()
        {
            CreatedOn = DateTime.Now;
            Snippets = new List<Snippet>();
        }

        public string Name { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
        public List<Snippet> Snippets { get; set; }

        public DateTime CreatedOn { get; set; }
        public DateTime? Modified { get; set; }
    }
}
